exporter/exporterhelper/internal/queue.go

```diff
@@ -0,0 +1,28 @@
+// Copyright The OpenTelemetry Authors
+// Copyright (c) 2019 The Jaeger Authors.
+// Copyright (c) 2017 Uber Technologies, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package internal // import "go.opentelemetry.io/collector/exporter/exporterhelper/internal"
+
+import (
+	"go.opentelemetry.io/collector/component"
+)
+
+// Queue defines a queue which can be backed by e.g. the memory-based ring buffer queue
+// (boundedMemoryQueue) or via a disk-based queue (persistentQueue)
+type Queue interface {
+	component.Component
+	// Offer inserts the specified element into this queue if it is possible to do so immediately without violating
+	// capacity restrictions, returning true upon success and false if no space is currently available.
+	Offer(item Request) bool
+	// Poll retrieves and removes the head of this queue.
+	Poll() <-chan Request
```

Comment:
The name seems a bit confusing, given that it returns a channel. Maybe `Chan()`, `Out()` or something like that?